### PR TITLE
Rework FaaS semantic conventions.

### LINKED ISSUE #1280: Introduce a new resource attribute for AWS Lambda function alias
**What are you trying to achieve?**

I propose adding a new optional resource attribute that will store the AWS Lambda function alias (if it's used at all), named `aws.lambda.alias`.

**Additional context.**

* [AWS Lambda alias docs](https://docs.aws.amazon.com/lambda/latest/dg/configuration-aliases.html)

### PR DESCRIPTION
**Pull request by Oberon00**
Fixes #1280 (through introduction of aws.lambda.invoked_arn span attribute, @mateuszrzeszutek please check if that is OK).
Addresses #1778.

## Changes

- More clearly describe some FaaS attributes, in particular ID-related ones.
- Add `aws.lambda.invoked_arn` and clarify that alias must be stripped from `faas.id` resource attribute for AWS Lambda.
- Describe what IDs to use for Azure and GCP (Note: It might be impractical to collect these IDs client-side).

CC @skonto

### CODE REVIEW

File: semantic_conventions/resource/faas.yaml [hunk 1]
```diff
@@ -8,32 +8,50 @@ groups:
         type: string
         required: always
```
Comment (carlosalberto) [inline semantic_conventions/resource/faas.yaml:9]: Not required? The paragraph below seems to indicate we may not be able to make it required.

Comment (Oberon00) [reply semantic_conventions/resource/faas.yaml:9]: The paragraph below is about the `id` for which I removed the `required`. But just the name should be available anywhere.

File: semantic_conventions/resource/faas.yaml [hunk 2]
```diff
@@ -8,32 +8,56 @@ groups:
...
-          [ARN](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html)
-          value, in GCP to the URI of the resource, and in Azure to the
-          [FunctionDirectory](https://github.com/Azure/azure-functions-host/wiki/Retrieving-information-about-the-currently-running-function)
-          field.
+          The unique ID of the single function that this runtime instance executes.
+        note: |
+          Depending on the cloud provider, use:
+
+          * **AWS Lambda:** The function [ARN](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html).
+          Take care not to use the "invoked ARN" directly but replace any
```
Comment (anuraaga) [inline semantic_conventions/resource/faas.yaml:27]: Do you know if this is possible?

Comment (Oberon00) [reply semantic_conventions/resource/faas.yaml:27]: The string replacement or that "the  same runtime instance may be invokable with multiple different aliases"? Yes, I verified both. You can easily produce that scenario with the Lambda console UI by creating two different aliases for `$LATEST` and then you can test both and you will see that the logs end up in the same logstream without restart.

You can even test it without creating an extra alias. E.g. using `aws lambda invoke --function-name 'arn:aws:lambda:us-east-1:12345:function:myfunction:$LATEST` will end up with a different invoked ARN than just `aws lambda invoke --function-name 'arn:aws:lambda:us-east-1:12345:function:myfunction`, but the same invoked version & log stream.

**Review by thisthat — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Comment by carlosalberto** [comment]
@mateuszrzeszutek Please review whenever you have cycles ;)

**Comment by skonto** [comment]
Thanks @Oberon00 looks good to me. I will adjust #1736 as needed.

File: specification/trace/semantic_conventions/instrumentation/aws-lambda.md [hunk 3]
```diff
@@ -35,9 +35,30 @@ Lambda `Context`.
...
 The following attributes SHOULD be set:
 
-- [`faas.execution`](../faas.md) - The value of the AWS Request ID, which is always available through an accessor on the Lambda `Context`
-- [`faas.id`](../../../resource/semantic_conventions/faas.md) - The value of the invocation ARN for the function, which is always available through an accessor on the Lambda `Context`
-- [`cloud.account.id`](../../../resource/semantic_conventions/cloud.md) - In some languages, this is available as an accessor on the Lambda `Context`. Otherwise, it can be parsed from the value of `faas.id` as the fifth item when splitting on `:`
+- [`faas.execution`][faas] - The value of the AWS Request ID, which is always available through an accessor on the Lambda `Context`.
+- [`faas.id`][faasres] - The value of the invocation ARN
+  for the function, which is always available through an accessor on the
+  Lambda `Context`, modified as follows: Discard all parts beyond the seventh (when split on `:`;
+  the seventh part is the function name) and append the [`faas.version`][faasres], separated by a colon.
```
Comment (mateuszrzeszutek) [inline specification/trace/semantic_conventions/instrumentation/aws-lambda.md:42]: 👍

**Review by mateuszrzeszutek — APPROVED** [review]
LGTM 👍

**Comment by carlosalberto** [comment]
@Oberon00 Please update your branch so we can merge this 👍

**Pull request merged**